backend/Sources/backend/Controllers/MeetupController.swift

```diff
@@ -0,0 +1,165 @@
+//
+//  MeetupController.swift
+//  backend
+//
+//  Created by Mauricio Cardozo on 9/28/25.
+//
+
+import CocoaHeadsCore
+import SwiftSoup
+import Vapor
+
+struct MeetupController: RouteCollection {
+  func boot(routes: any Vapor.RoutesBuilder) throws {
+    let meetupRoutes = routes.grouped("meetup")
+    meetupRoutes.post(use: self.event)
+  }
+
+  @Sendable
+  func event(req: Request) async throws -> MeetupEvent {
+    let meetup = try req.content.decode(MeetupEventRequest.self)
+
+    guard
+      let url = URL(string: meetup.url),
+      case (let data, _) = try await URLSession.shared.data(from: url),
+      let html = String(data: data, encoding: .utf8)
+    else {
+      throw Error.urlError
+    }
```

Comment:
I suggest you to split into two different `guard`s here. You are not being clear enough when raising errors.
If something goes wrong with the URL, then you will throw `Error.urlError`, however, in case you get an error while you are requesting the content from that URL using URLSession or get an error while you are converting from the received data into a String, you will also throw `Error.urlError`. You had to separate at least in two more guards to throw the proper error.